### DISCUSSION TITLE
What is your opinion on using adjacent or surrounding objects for filtering?

### DISCUSSION THREAD

**Discussion by DerDings** [streetcomplete/StreetComplete#3044] [Ideas]
There are already some quests which make use of surrounding elements for filtering. One example is the cycleway quest, which excludes roads where a cycle-able path is near and aligned.

What do you think of using similar logic also for other quests? Now that quests are created locally and not via overpass anymore, this should have become simpler and faster.
I have a lot of ideas on how to improve certain filters by having a look at other elements. Most of them are even much simpler than the isNearAndAligned stuff. But before I put effort in refining those ideas, testing them with example overpass queries and opening issues, I'd like to ask if this approach is desired.
The use case I see is mainly the filters which are currently very restrictive to avoid spam, so they do also miss some good quest candidates.

**Comment by smichel17** [comment]
Note: despite having the data locally, these tend to be resource-intensive to process. See https://github.com/streetcomplete/StreetComplete/issues/2803, for example, which was a result of some relations being really huge.

> before I put effort in refining those ideas

I suspect it will be on a case-by-case basis. So what *I* would do is post a list of ideas, see which ones there is more interest in, and only put in effort to refine those.

**Reply by Helium314** [reply]
I remember performance was the reason not to exclude bus stops with nearby waste bins from the waste bin quest, so I assume there has to be significant benefit from checking surrounding elements

**Comment by andrewharvey** [comment]
To be honest I think we should go in the other direction use little to none of the surrounding objects to decide which quests to ask, and instead just always ask the quest.

A road might have both an on-road lane or other infrastructure plus there might be an adjacent off road track, SC would never know unless it asks.

**Comment by DerDings** [comment]
So most comments are about nearby objects. What about checking for adjacent objects?
There are two kinds of this: 
A: Checking for connected things, like roads which share a node with a certain other kind of highway
B: Checking for intersecting ways without connection, like the maxheight quest does.
Do those kind of filters also have such a negative impact on performance?
